### ISSUE TITLE
Wallet sync stops on Windows 10 when user connects/disconnects USB peripherals.

### ISSUE THREAD

**Issue by ggcaponetto** [input-output-hk/daedalus#2519]
### Expected behaviour

I expect the Daedalus wallet to perform a full synchronization even if i connect or disconnect the mouse/keyboard usb peripheral during synchronization.

### Current behaviour

While the wallet is synchronizing and the initial loading screen is displayed, Daedalus crashes if I unplug and replug the USB mouse and keyboard. From this point on Daedalus is not able to perform the synchronization after a restart and always gets stuck at the same percentage. The only way to perform a synchronization is to manually delete the wallet and chain data in `%APPDATA%` and restart Daedalus.

#### OS
Windows 10 Pro
Version: 20H2
Build: 19042.867

#### Daedalus
Version: 4.0.4

**Comment by miorsufianiohk** [comment]
Thank you for contacting us @ggcaponetto . We will have this check and get back to you.

**Comment by miorsufianiohk** [comment]
Hi @ggcaponetto ,

I hope that you are well. 

We unsuccessfully tried to replicate the issue that you raised, unfortunately. However, the issue might be related to another issue that has been found (which also relates to usb device crashing Daedalus on Windows). I could see that this issue is already on the development pipeline to be looked at in a few weeks time. 

In the meantime, could you please give further details. Is this issue a blocker for you from using Daedalus? After deleting the wallet and chain folders and resynchronising, do you have to repeat the process again?

I look forward to receiving your reply in due course.

Many thanks,
Mior

**Comment by ggcaponetto** [comment]
Thank you for trying. This issue does not block me from using daedalus at the moment. I will provide you the logs this weekend.

The steps for a successful recover are:

1. Opening Daedalus after it crashed
2. Shutting down the node (Exiting the Deaedalus app)
2. Deleting the wallet and chain data
3. Rebooting the machine
4. Open Daedalus

**Comment by nikolaglumac** [comment]
@ggcaponetto do you maybe use a YubiKey?

**Comment by nikolaglumac** [comment]
Also, an exact model of your USB devices that trigger this issue would be appreciated 🙏

**Comment by ggcaponetto** [comment]
Hi.
The mouse that i use i a [Microsoft Sculpt Mobile Mouse](https://www.microsoft.com/accessories/en-my/products/mice/sculpt-mobile-mouse/43u-00005) (Model 1569). Other mice i have work fine and I don't have any YubiKey connected.

My system is:

````
{"at":"2021-04-17T08:23:32.757Z","env":"mainnet:Windows:10.0.19042","ns":["daedalus","v4.0.5","*mainnet*"],"data":{"cardanoNodeVersion":"1.25.1","cardanoWalletVersion":"2021.3.4","cpu":[{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":13975296,"nice":0,"sys":86250265,"idle":104564828,"irq":7134500}},{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":13854562,"nice":0,"sys":41313265,"idle":149622296,"irq":1041140}},{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":14859250,"nice":0,"sys":59941250,"idle":129989625,"irq":4870312}},{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":14030687,"nice":0,"sys":46766421,"idle":143993015,"irq":1541171}},{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":11103375,"nice":0,"sys":41643265,"idle":152043453,"irq":1839046}},{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":9215375,"nice":0,"sys":37797328,"idle":157777390,"irq":974187}},{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":7092156,"nice":0,"sys":10792656,"idle":186905265,"irq":487171}},{"model":"AMD FX(tm)-8350 Eight-Core Processor           ","speed":4013,"times":{"user":5524250,"nice":0,"sys":7733828,"idle":191532000,"irq":162640}}],"daedalusVersion":"4.0.5","isBlankScreenFixActive":false,"network":"mainnet","osName":"Windows","platformVersion":"10.0.19042","ram":"17073049600","startTime":"2021-04-17T08:23:32.757Z"},"app":["daedalus"],"msg":"Updating System-info.json file","pid":"","sev":"info","thread":""}
````

Unfortunately I couldn't spot any error/crash log entry.

**Comment by nikolaglumac** [comment]
So the only way to reproduce the issue is to use that specific type of mouse?
The issue does not happen when other mice are used nor with any other USB hardware like the keyboard for example?

**Comment by ggcaponetto** [comment]
Exactly. It seems to be an issue specific to that mouse but mybe others too. It wouldn't be a problem at all if only Daedalus was able to recover itself and resync without having to delete wallet and chain data manually.

**Comment by Scalextrix** [comment]
I am having this same issue: https://github.com/input-output-hk/daedalus/issues/2527

After reading above, in my case it is because of a Yubikey, with the Yubikey plugged into the keyboard USB port, Daedalus crashes on disconnect.  If the Yubikey is removed first, then disconnect does not cause a Daedalus crash.

**Comment by nikolaglumac** [comment]
@gabriela-ponce @ManusMcCole can you please try to reproduce this ☝️

**Comment by Scalextrix** [comment]
After a little more experimentation, plugging the Yubikey into any of several USB ports (peripheral or not) also crashes the Daedalus wallet.  Basically any addition or removal of the Yubikey causes the issue.

**Comment by ManusMcCole** [comment]
Hi @nikolaglumac. I was able to reproduce the error with the Yubi keys. Plugging in or removing the key crashed Daedalus immediately. When the Yubi key was unplugged this error was shown [screenshot](https://drive.google.com/file/d/1yvjr0PQRzaTAey9F1GE2Jd1-sbgLBZ9Z/view). When it is plugged in Daedalus crashes with no error message.I was not able to recreate Daedalus crashing when a wireless mouse was plugged/unplugged. My wireless mouse brand is Patuoxun 2.4G USB Wireless Mice Optical. Perhaps @gabriela-ponce will be able to try another brand

**Comment by gabriela-ponce** [comment]
@nikolaglumac I can't reproduce the issue with the USB peripherals. I don't own the device mentioned in [this comment](https://github.com/input-output-hk/daedalus/issues/2519#issuecomment-821790599), but tried with other USB devices, including two different wireless mice. 
However, Daedalus closes for me just after plugging the Yubikey (5C NFC).

**Comment by hyliu1989** [comment]
This happens to me too. The steps that lead to my case were
1. I used a Kensington trackball to double click the icon to open the Daedalus Mainnet from the Desktop.
2. I then disconnected that trackball and waited for 5 seconds
3. Daedalus crashed.
I bet that every mouse (other than the Kensington trackball) might trigger the case. The Kensington trackball was the only mouse I have when running the step above.

In addition, I tried another case with two mouses (A and B). 
1. I used mouse A to click the icon on the Desktop and disconnected mouse A, while leaving mouse B connected; Daedalus did not like this and it crashed. 
2. I then reconnected both mouses back. 
3. The second time I used mouse B to click the icon on the Desktop but then disconnected mouse A; Daedalus seemed to be ok with this arrangement.

Daedalus version was 4.10. The operating system was Windows 10.

**Comment by nikolaglumac** [comment]
This issue has been fixed in https://github.com/input-output-hk/daedalus/pull/2673
Is to be released in the next Daedalus version.

**Comment by disassembler** [comment]
Closing as part of issue backlog cleanup — issues older than 90 days are being retired to give the new team a clean slate. Please open a new issue if this is still relevant.
